packages/actions-server/src/ActionsProxy.test.ts

```diff
@@ -0,0 +1,279 @@
+        parameters: Record<string, unknown>;
+        requestId: string;
+    };
+}
+
+describe("ActionsProxy", () => {
+    let server: ActionsProxy;
+    let client: McpClient;
+    let clientId: string;
+    let wsPort: number;
+
+    beforeEach(async () => {
+        // Use a random port between 30000-40000
+        wsPort = Math.floor(Math.random() * 10000) + 30000;
+        server = new ActionsProxy(wsPort);
+
+        // Register tools before connecting
+        clientId = await server.registerTools([
+            {
+                name: "testTool",
+                description: "A test tool",
+                parameters: z.object({
+                    input: z.string()
+                })
+            }
+        ]);
+
+        // Create client and connect
+        const [clientTransport, serverTransport] = InMemoryTransport.createLinkedPair();
+        client = new McpClient({
+            name: "test-client",
+            version: "1.0.0",
+            transport: clientTransport
+        });
+
+        // Connect after registering tools
+        await Promise.all([
+            client.connect(clientTransport),
+            server.getMcpServer().server.connect(serverTransport)
+        ]);
+    });
+
+    afterEach(async () => {
+        server.close();
+    });
+
+    it("should register tools and allow MCP client to call them", async () => {
+        // Connect WebSocket with clientId
+        const ws = new WebSocket(`ws://localhost:${wsPort}?clientId=${clientId}`);
+        await new Promise<void>((resolve) => ws.on("open", resolve));
+
+        // Set up message listener for TOOL_REQUEST
+        const requestPromise = new Promise<ToolRequest>((resolve) => {
+            ws.on('message', (data: Buffer) => {
+                const message = JSON.parse(data.toString());
+                if (message.type === 'TOOL_REQUEST') {
+                    resolve(message);
+                }
+            });
+        });
+
+        // Wait for client to be connected
+        await new Promise<void>((resolve) => {
+            const checkConnection = () => {
+                if (server.isClientConnected(clientId)) {
+                    resolve();
+                } else {
+                    setTimeout(checkConnection, 100);
+                }
+            };
+            checkConnection();
+        });
+
+        // Call the tool
+        const toolCallPromise = client.callTool({
+            name: "testTool",
+            arguments: {
+                clientId,
+                clientSecret: "test-secret",
+                input: "test input"
+            }
+        });
+
+        // Wait for TOOL_REQUEST
+        const request = await requestPromise;
+
+        // Send TOOL_RESPONSE
+        ws.send(JSON.stringify({
+            type: "TOOL_RESPONSE",
+            payload: {
+                requestId: request.payload.requestId,
+                result: { success: true, input: request.payload.parameters.input }
+            }
```

Comment:
This is a mock response for our test. In our application, the `ActionsManager` would execute the client-side tool like it does now and return the result.